### PR TITLE
Work around LD_PRELOAD error on reMarkable 2 devices

### LINKED ISSUE #10485: Crash when attempting to connect to Calibre server
* KOReader version: 2023.04
* Device: reMarkable 2 (2.15.1.1189)

#### Issue

When attempting to connect to a Calibre server, KOReader crashes with the message "attempt to perform arithmetic on a string value" [here](https://github.com/koreader/koreader/blob/v2023.04/frontend/util.lua#L857). (The below logs show line 860 because I added some debug logging to the code.) In my case, the value of `stage[k]` is `ERROR`.

KOReader's logs are attached below. At the same time, Calibre also displays a pop-up with the following log:

<details>
  <summary>Calibre log</summary>

```
calibre, version 6.17.0
ERROR: Error: Error communicating with device

Device responded with incorrect information

Traceback (most recent call last):
  File "calibre/gui2/device.py", line 104, in run
  File "calibre/gui2/device.py", line 522, in _get_device_information
  File "calibre/devices/smart_device_app/driver.py", line 59, in _synchronizer
  File "calibre/devices/smart_device_app/driver.py", line 1282, in free_space
  File "calibre/devices/smart_device_app/driver.py", line 59, in _synchronizer
  File "calibre/devices/smart_device_app/driver.py", line 653, in _call_client
  File "calibre/devices/smart_device_app/driver.py", line 695, in _receive_from_client
calibre.devices.errors.ControlError: Device responded with incorrect information
```
</details>

#### Steps to reproduce

1. Connect a reMarkable 2 with KOReader via USB to a device with Calibre
2. Open Calibre
3. Start Calibre's device connection server if not configured to do so automatically
4. Open KOReader
5. Within KOReader, tap on the tools icon, then on "Calibre" -> "Connect"

##### `crash.log` (if applicable)

<details>
  <summary>crash.log</summary>

```
Original fb settings: bitdepth = 16, rotation = 0
Switching fb bitdepth to 8bpp & rotation to Portrait
[FBInk] Detected a reMarkable 2 (Zero Sugar)
[FBInk] This device does not support HW inversion
[FBInk] Running under the rm2fb compatibility shim (version 0.1), functionality may be limited
[FBInk] Clock tick frequency appears to be 100 Hz
[FBInk] Screen density set to 226 dpi
[FBInk] Variable fb info: 1404x1872, 16bpp @ rotation: 0 (Upright, 0°)
[FBInk] Fixed fb info: ID is "mxcfb", length of fb mem: 5256576 bytes & line length: 2808 bytes
[FBInk] Fontsize set to 24x24 (IBM base glyph size: 8x8)
[FBInk] Line length: 58 cols, Page size: 78 rows
[FBInk] Pen colors set to #000000 for the foreground and #FFFFFF for the background
[FBDepth] Screen is 1404x1872 (1404x1872 addressable, fb says 1404x1872)
[FBDepth] Buffer is mapped for 5256576 bytes with a scanline stride of 2808 bytes
[FBDepth] Switching fb to 8bpp @ rotation 1 . . .
Updating bitdepth from 16bpp to 8bpp
Sanitizing grayscale flag for 8bpp
Updating rotation from 0 (Upright, 0°) to 1 (Clockwise, 90°)
Bitdepth is now 8bpp (grayscale: 1) @ rotate: 1 (Clockwise, 90°)
[FBDepth] Screen is 1404x1872 (1404x1872 addressable, fb says 1404x1872)
[FBDepth] Buffer is mapped for 5256576 bytes with a scanline stride of 2808 bytes
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 05/25/23-15:56:10
 [*] Version: v2023.04

ffi.load: libs/libutf8proc.so.2
ffi.load: blitbuffer
05/25/23-15:56:10 INFO  Starting reMarkable 2.0 
05/25/23-15:56:10 INFO  Oxide running?: false 
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1' from LD_PRELOAD cannot be preloaded (internal error): ignored.
[ko-input] Forked off fake event generator (pid: 17928)
05/25/23-15:56:10 INFO  initializing for device reMarkable 2.0 
05/25/23-15:56:10 INFO  framebuffer resolution: {
  h = 1872,
  w = 1404
} --[[table: 0x76672a70]] 
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
05/25/23-15:56:11 INFO  Loading plugins from directory: plugins 
ffi.load: sqlite3
05/25/23-15:56:19 INFO  calibre info loaded from disk in 33.000 milliseconds: 238 books. 0 pruned 
ffi.load: libs/libczmq.so.1
ffi.load: libs/libzmq.so.4
ffi.load: libs/libczmq.so.1
05/25/23-15:56:19 INFO  connecting to calibre @ 192.168.178.35:9090 
./luajit: frontend/util.lua:860: attempt to perform arithmetic on a string value
stack traceback:
        frontend/util.lua:860: in function 'diskUsage'
        plugins/calibre.koplugin/wireless.lua:43: in function 'getFreeSpace'
        plugins/calibre.koplugin/wireless.lua:474: in function 'getFreeSpace'
        plugins/calibre.koplugin/wireless.lua:336: in function 'onReceiveJSON'
        plugins/calibre.koplugin/wireless.lua:141: in function 'receiveCallback'
        frontend/ui/message/streammessagequeue.lua:86: in function '(for generator)'
        frontend/ui/uimanager.lua:1424: in function 'processZMQs'
        frontend/ui/uimanager.lua:1487: in function 'handleInput'
        frontend/ui/uimanager.lua:1559: in function 'run'
        ./reader.lua:319: in main chunk
        [C]: at 0x000140f1
!!!!
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Uh oh, something went awry... (Crash n°1: 05/25/23 @ 15:56:24)
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1' from LD_PRELOAD cannot be preloaded (internal error): ignored.
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1' from LD_PRELOAD cannot be preloaded (internal error): ignored.
Running on Linux 5.4.70-v1.2.5-rm11x (#1 SMP PREEMPT Sun Oct 23 20:48:32 UTC 2022)
Attempting to restart KOReader . . .
!!!!
---------------------------------------------
                launching...
  _  _____  ____                _
 | |/ / _ \|  _ \ ___  __ _  __| | ___ _ __
 | ' / | | | |_) / _ \/ _` |/ _` |/ _ \ '__|
 | . \ |_| |  _ <  __/ (_| | (_| |  __/ |
 |_|\_\___/|_| \_\___|\__,_|\__,_|\___|_|

 It's a scroll... It's a codex... It's KOReader!

 [*] Current time: 05/25/23-15:56:39
 [*] Version: v2023.04

ffi.load: libs/libutf8proc.so.2
ffi.load: blitbuffer
05/25/23-15:56:39 INFO  Starting reMarkable 2.0 
05/25/23-15:56:39 INFO  Oxide running?: false 
ERROR: ld.so: object '/opt/lib/librm2fb_client.so.1' from LD_PRELOAD cannot be preloaded (internal error): ignored.
[ko-input] Forked off fake event generator (pid: 17957)
05/25/23-15:56:39 INFO  initializing for device reMarkable 2.0 
05/25/23-15:56:39 INFO  framebuffer resolution: {
  h = 1872,
  w = 1404
} --[[table: 0x7655c7a0]] 
ffi.load: libs/libmupdf.so
ffi.load: libs/libwrap-mupdf.so
ffi.load: libs/libfreetype.so.6
ffi.load: libs/libharfbuzz.so.0
ffi.load: libs/libzstd.so.1
05/25/23-15:56:40 INFO  Loading plugins from directory: plugins 
ffi.load: sqlite3
05/25/23-15:56:49 INFO  no dialog left to show 
05/25/23-15:56:49 INFO  Tearing down UIManager with exit code: 0 
[ko-input] Closed input device with fd: 7 @ idx: 3
[ko-input] Closed input device with fd: 6 @ idx: 2
[ko-input] Closed input device with fd: 5 @ idx: 1
[ko-input] Closed input device with fd: 3 @ idx: 0
Restoring original fb bitdepth @ 16bpp & rotation @ 0
[FBInk] Detected a reMarkable 2 (Zero Sugar)
[FBInk] This device does not support HW inversion
[FBInk] Running under the rm2fb compatibility shim (version 0.1), functionality may be limited
[FBInk] Clock tick frequency appears to be 100 Hz
[FBInk] Screen density set to 226 dpi
[FBInk] Variable fb info: 1404x1872, 16bpp @ rotation: 0 (Upright, 0°)
[FBInk] Fixed fb info: ID is "mxcfb", length of fb mem: 5256576 bytes & line length: 2808 bytes
[FBInk] Fontsize set to 24x24 (IBM base glyph size: 8x8)
[FBInk] Line length: 58 cols, Page size: 78 rows
[FBInk] Pen colors set to #000000 for the foreground and #FFFFFF for the background
[FBDepth] Screen is 1404x1872 (1404x1872 addressable, fb says 1404x1872)
[FBDepth] Buffer is mapped for 5256576 bytes with a scanline stride of 2808 bytes
[FBDepth] Current bitdepth is already 16bpp!
[FBDepth] Current rotation is already 0!
```
</details>

### PR DESCRIPTION
**Pull request by TheLastZombie**
See https://github.com/koreader/koreader/issues/10485#issuecomment-1563875642 for details.

Fixes #10485.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10490)

### CODE REVIEW

File: frontend/util.lua [hunk 1]
```diff
@@ -859,7 +859,7 @@ end
 function util.diskUsage(dir)
     -- safe way of testing df & awk
     local function doCommand(d)
-        local handle = io.popen("df -k " .. d .. " 2>&1 | awk '$3 ~ /[0-9]+/ { print $2,$3,$4 }' 2>&1 || echo ::ERROR::")
+        local handle = io.popen("df -k " .. d .. " 2>&1 | awk '$3 ~ /[0-9]+/ { print $2,$3,$4 }' 2>&1 | grep -v 'ERROR: ld.so' || echo ::ERROR::")
```
Comment (Frenzie) [inline frontend/util.lua:862]: Wouldn't 'ignored' or something make more sense?

Comment (pazos) [reply frontend/util.lua:862]: or redirect errors to `/dev/null` in `df` output

Comment (TheLastZombie) [reply frontend/util.lua]: Redirecting is probably the cleanest reasonable solution. Though I had to do the same with awk, as it's launched with the same LD_PRELOAD variable.

**Review by NiLuJe — APPROVED** [review]

File: frontend/util.lua (hunk 1 above)
Comment (NiLuJe) [reply frontend/util.lua]: FWIW, we could also filter out stuff from the env via `env`, but void'ing stderr is more generic ;).

**Review by Frenzie — APPROVED** [review]

**Comment by NiLuJe** [comment]
I'd wager some other stuff may exhibit similar issues, so I'd ask the Toltec guys what's up with that, but for now, this looks good ;).

**Pull request merged**